wizard/WizardManageWalletUI.qml

```diff
@@ -45,15 +51,45 @@ Item {
     property alias restoreHeightVisible: restoreHeightItem.visible
     property alias walletName : accountName.text
     property alias progressDotsModel : progressDots.model
+    property alias recoverFromKeysAddress: addressLine.text;
+    property alias recoverFromKeysViewKey: viewKeyLine.text;
+    property alias recoverFromKeysSpendKey: spendKeyLine.text;
+    // recover mode or create new wallet
+    property bool recoverMode: false
+    // Recover form seed or keys
+    property bool recoverFromSeedMode: true
 
-    // TODO extend properties if needed
+    function checkFields(){
+        var addressOK = walletManager.addressValid(addressLine.text, wizard.settings.testnet)
+        var viewKeyOK = walletManager.keyValid(viewKeyLine.text, addressLine.text, true, wizard.settings.testnet)
+        // Spendkey is optional
+        var spendKeyOK = (spendKeyLine.text.length > 0)? walletManager.keyValid(spendKeyLine.text, addressLine.text, false, wizard.settings.testnet) : true
 
-    anchors.fill: parent
-    Row {
+        addressLine.error = !addressOK && addressLine.text.length != 0
+        viewKeyLine.error = !viewKeyOK && viewKeyLine.text.length != 0
+        spendKeyLine.error = !spendKeyOK && spendKeyLine.text.length != 0
+
+        return addressOK && viewKeyOK && spendKeyOK
+    }
+
+    function checkNextButton(){
+        wizard.nextButton.enabled = false
+        console.log("check next", recoverFromSeed.visible)
+        if(recoverMode && !recoverFromSeedMode) {
+            console.log("checking key fields")
+            wizard.nextButton.enabled = checkFields();
+        } else if (recoverMode && recoverFromSeedMode) {
+            console.log("Checking seed")
+            var wordsArray = Utils.lineBreaksToSpaces(uiItem.wordsTextItem.memoText).split(" ");
+            wizard.nextButton.enabled = wordsArray.length === 25;
```

Comment:
I'd put this check in a separate function (ie, is_seed_ready or something) so it's obvious, as it seems likely to change next time we add things to the seed (eg, creation date, etc).